woocommerce/woocommerce-gutenberg-products-block
# Add Google Analytics Product Block and Checkout Events

This PR adds events for product block views, add to cart links, search functionality, exceptions, and checkout. It demonstrates how actions can be used from blocks themselves, or triggered by inline JS in the case of SSR and non-dynamic blocks.

I've moved the GA code to it's own utility class now that more PHP side code is needed.

Closes #3935
Closes #3951
Closes #3959

### How to test the changes in this Pull Request:

See the setup instructions in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3953. Please ensure the Google Analytics Browser Extension is not in 'debug mode' as this will cause events to not fire.

1. View a single product block. Check a view item event fires.
2. Click the add to cart button in single product block. Check add to cart event fires.
3. Search for something using the product search block. Check for the event.

To test the checkout functionality,

1. Visit checkout with items in your cart. You will see a begin_checkout event
2. Start editing fields. Depending on the field you edit (which step) you will see a checkout_progress event. e.g. adding billing information will add checkout_progress with a number indicating the step you are on.
3. Select a shipping rate. See that a set_checkout_option event fired.
4. Select a payment method. See that a set_checkout_option fired.
5. Leave a required field blank and place order. See _exception_ event.
6. Place order. See add_payment_info event.

To test SSR blocks:

1. Insert product grid block into a page
2. View the page. Check a view_item_list event fired in GA.
